e2e/validation.go

```diff
@@ -0,0 +1,117 @@
+package e2e_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+	"testing"
+
+	"github.com/Azure/agentbakere2e/scenario"
+)
+
+var (
+	sysctlsAlwaysSet = []string{
+		"net.ipv4.tcp_retries2",
+		"net.core.message_burst",
+		"net.core.message_cost",
+		"net.core.somaxconn",
+		"net.ipv4.tcp_max_syn_backlog",
+		"net.ipv4.neigh.default.gc_thresh1",
+		"net.ipv4.neigh.default.gc_thresh2",
+		"net.ipv4.neigh.default.gc_thresh3",
+	}
+)
+
+func runLiveVMValidators(ctx context.Context, t *testing.T, vmssName, sshPrivateKey string, opts *scenarioRunOpts) error {
+	privateIP, err := getVMPrivateIPAddress(ctx, opts.cloud, opts.suiteConfig.subscription, *opts.chosenCluster.Properties.NodeResourceGroup, vmssName)
+	if err != nil {
+		return fmt.Errorf("unable to get private IP address of VM on VMSS %q: %s", vmssName, err)
+	}
+
+	podName, err := getDebugPodName(opts.kube)
+	if err != nil {
+		return fmt.Errorf("unable to get debug pod name: %s", err)
+	}
+
+	validators := commonLiveVMValidators()
+	if opts.scenario.LiveVMValidators != nil {
+		validators = append(validators, opts.scenario.LiveVMValidators...)
+	}
+
+	for _, validator := range validators {
+		desc := validator.Description
+		command := validator.Command
+		log.Printf("running live VM validator: %q", desc)
+
+		execResult, err := pollExecOnVM(ctx, opts.kube, privateIP, podName, sshPrivateKey, command)
```

Comment:
hmm. this makes sense to me. wondering if there is a way to simplify it a bit.

it almost feels like we could abstract the "run a command on the VM" part into a struct or higher level function, and "build" the validator, then "run" (execute/assert) it

I'm okay with this, definitely a net win overall, but curious about structure

Reply:
yeeeeah I was kinda thinking the same thing, I like the idea of introducing a struct along the lines of "RemoteCommandOpts" that would contain the necessary config (private key, private IP, pod name, etc.) which could then implement a receiver that actually takes in a command and executes it

might be misunderstanding what you're saying tho

Reply:
either way, maybe a good follow-up item